### PR TITLE
Move to OCS API

### LINKED ISSUE #48: Document API
Please document API
- mark file as locked (preferred via webdav)
- mark file as unlocked
- propfind on webdav
  - as  discussed we need two different file locking scopes (maybe  local and online?)
- is there a capability?
-

### PR DESCRIPTION
**Pull request by juliushaertl**
This will make sure that the API can be used properly by clients.

Fixes #48 though the final API to be implemented in clients should also include further API adjustments for #51, so this might change during further development effort.

### CODE REVIEW

File: lib/Controller/LockController.php [hunk 1]
```diff
@@ -47,7 +46,7 @@
  *
  * @package OCA\FilesLock\Controller
  */
-class LockController extends Controller {
+class LockController extends \OCP\AppFramework\OCSController {
```
Comment (ArtificialOwl) [inline lib/Controller/LockController.php:49]: maybe import `OCSController`

File: lib/Controller/LockController.php [hunk 2]
```diff
@@ -94,7 +93,7 @@ public function locking(string $fileId): DataResponse {
 			$user = $this->userSession->getUser();
 			$file = $this->fileService->getFileFromId($user->getUID(), (int)$fileId);
 
-			$lock = $this->lockService->lockFile($file, $user);
+			$lock = $this->lockService->lockFileAsUser($file, $user);
```
Comment (ArtificialOwl) [inline lib/Controller/LockController.php:96]: I haven't test does `lockFileAsUser()` exist in `LockService` ?

**Comment by ArtificialOwl** [comment]
Code looks good, will test (the OCS part) later

File: lib/Controller/LockController.php (hunk 2 above)
Comment (juliushaertl) [reply lib/Controller/LockController.php:96]: Whoops, good catch, seems I failed to commit only the relevant parts here, this is from #54

**Comment by ArtificialOwl** [comment]
can you remove it so we can work on merging this one ?

**Comment by juliushaertl** [comment]
Comments addressed

**Review by ArtificialOwl — APPROVED** [review]

**Pull request merged**